jax/_src/numpy/lax_numpy.py

```diff
@@ -1859,6 +1863,7 @@ def concatenate(arrays: np.ndarray | Array | Sequence[ArrayLike],
 @util._wraps(np.vstack)
 def vstack(tup: np.ndarray | Array | Sequence[ArrayLike],
            dtype: DTypeLike | None = None) -> Array:
+  arrs: Array | list[Array]
```

Comment:
Annotate in anticipation of using the variable one of two ways.  Otherwise, the type checker treats the first use as an implicit annotation, and assumes too narrow a type.